docs/reference-use-cases.md

```diff
@@ -0,0 +1,215 @@
+
+- As an operator, I expect the Cluster API to maintain the number and type of Nodes that I have currently requested as members of the cluster.
+
+### Configuration Updates
+
+- As an operator, given that I have a management cluster and a workload cluster, I want to update the IaaS credentials used to lifecycle manage my workload cluster because the correct credentials have changed.
+
+- As an operator, given I have a management cluster and a workload cluster, I want to change the “minor customizations” that I’m applying before kubelet starts.
+
+- As an operator, given that I have deployed a workload cluster via Cluster API, I want to change config in my workload cluster for which Cluster API is authoritative and have a Cluster API controller manage the deployment of that new configuration over the workload cluster.
+
+- As an operator, given that I have deployed a workload cluster via Cluster API and used Cluster API controller to manage the deployment of a new (broken) configuration, I want to revert to the previous working configuration.
+
+- As an operator, I want to declare the attributes (os, kernel, CRI)  of the nodes I want my workload to run on. The CAPI provider/controller should select an appropriate image that satisfies my constraints/attributes.
+    - If the provider does not support the attributes I have specified or find an appropriate image it should fail with an appropriate error.
+
+### Security
+
+- As an operator, given I have a management cluster and a workload cluster, I want to know when my cluster’s/machines’ certificates will expire, so that I can plan to rotate them.
+
+- As an operator, given I have a management cluster and a workload cluster, I want to automatically, periodically repave all the nodes of my cluster to reduce the risk of unauthorized software running on my machines.
+
+- As an operator, I want an external CA to sign certificates for the workload cluster control plane.
+
+- 🔭As an operator, given I have a management cluster and a workload cluster, I want to rotate all the certificates and credentials my machines are using.
+    - Some certificates might get rotated as part of machine upgrade, but are otherwise the above is out of scope.
+
+- 🔭 As an operator, given I have a management cluster and a workload cluster, I want to rotate/change the CA used to sign certificates for my workload cluster.
+
+- 🔭 As an operator, I want an external CA to sign certificates for workload cluster kubelets.
+
+### Upgrades
+- As an operator, given I have a management cluster and a workload cluster, I want to patch the OS running on all of my machines (e.g. for a CVE).
```

Comment:
should this be scope of cluster-api? I assume this is job of cloud-admin (off-premise and on-premise)?
after all, cluster api should not see the operating system it's running on and should not be authorized to do so
otherwise this might be a security hole?

Reply:
It's probably ok to keep it, patching could mean (in some form) to roll out an image upgrade